### PR TITLE
Feat PUD-975 (pci.storage.databases): add service integration tab

### PR DESCRIPTION
**Pull request by jperchoc**
PUD-975

| Question         | Answer
| ---------------- | ---
| Branch?          | `feat/PUD-819`
| Bug fix?         | no
| New feature?     | yes
| Breaking change? | no
| Tickets          | PUD-975
| License          | BSD 3-Clause

- [ ] Try to keep pull requests small so they can be easily reviewed.
- [ ] Commits are signed-off
- [ ] Only FR translations have been updated
- [ ] Branch is up-to-date with target branch
- [ ] Lint has passed locally
- [ ] Standalone app was ran and tested locally
- [ ] Ticket reference is mentioned in linked commits (internal only)
- [ ] Breaking change is mentioned in relevant commits

## Description

Add Service integration tab

## Related
```
PUD-833 <-- PUD-819
               ^
               |-- PUD-975 <-- PUD-972 
				^
				|-- PUD-969
				|-- PUD-973 -- PUD-1046 -- PUD-1057 -- PUD-1047
				|-- PUD-1007
```
[PUD-833](https://github.com/ovh/manager/pull/5605)
[PUD-819](https://github.com/ovh/manager/pull/5747)
[PUD-975](https://github.com/ovh/manager/pull/5765)
[PUD-972](https://github.com/ovh/manager/pull/5748)
[PUD-969](https://github.com/ovh/manager/pull/5774)
[PUD-973](https://github.com/ovh/manager/pull/5749)
[PUD-1007](https://github.com/ovh/manager/pull/5769)
[PUD-1046](https://github.com/ovh/manager/pull/5865)
[PUD-1057](https://github.com/ovh/manager/pull/5872)
[PUD-1047](https://github.com/ovh/manager/pull/5873)

### CODE REVIEW

File: packages/manager/modules/pci/src/projects/project/storages/databases/database/service-integration/add/add.html [hunk 1]
```diff
@@ -0,0 +1,24 @@
+<oui-modal
+    ng-form="addServiceIntegration"
```
Comment (mohammed-zahaf) [inline packages/manager/modules/pci/src/projects/project/storages/databases/database/service-integration/add/add.html:2]: I think you missed to create a HTML form element.
Or it's declared at parent template ?

File: packages/manager/modules/pci/src/projects/project/storages/databases/database/service-integration/add/translations/Messages_fr_FR.json [hunk 2]
```diff
@@ -0,0 +1,8 @@
+{
+  "pci_databases_service_integration_add_confirm": "Confirmer",
+  "pci_databases_service_integration_add_cancel": "Annuler",
+  "pci_databases_service_integration_add": "Ajouter un Kafka",
```
Comment (mohammed-zahaf) [inline packages/manager/modules/pci/src/projects/project/storages/databases/database/service-integration/add/translations/Messages_fr_FR.json:4]:
```suggestion
  "pci_databases_service_integration_add": "Ajouter un service Kafka",
```

File: packages/manager/modules/pci/src/projects/project/storages/databases/database/service-integration/delete/translations/Messages_fr_FR.json [hunk 3]
```diff
@@ -0,0 +1,8 @@
+{
+  "pci_databases_service_integration_delete_confirm": "Confirmer",
+  "pci_databases_service_integration_delete_cancel": "Annuler",
+  "pci_databases_service_integration_delete": "Supprimer un service Kafka",
+  "pci_databases_service_integration_delete_description": "Vous allez supprimer votre service Kafka. Souhaitez-vous continuer ?",
+  "pci_databases_service_integration_delete_success_message": "Votre service Kafka {{service}} a été supprimé avec succès",
+  "pci_databases_service_integration_delete_error_message": "Une erreur est survenue lors de la suppression de votre service: {{message}}"
```
Comment (mohammed-zahaf) [inline packages/manager/modules/pci/src/projects/project/storages/databases/database/service-integration/delete/translations/Messages_fr_FR.json:7]:
```suggestion
  "pci_databases_service_integration_delete_error_message": "Une erreur est survenue lors de la suppression de votre service : {{message}}"
```

File: packages/manager/modules/pci/src/projects/project/storages/databases/database/service-integration/service-integration.routing.js [hunk 4]
```diff
@@ -0,0 +1,127 @@
...
+            projectId,
+            database.engine,
+            database.id,
+          ).then((integrations) => {
+            const serviceIntegrations = map(integrations, (i) => {
+              const serviceIntegration = new ServiceIntegration(i);
+              serviceIntegration.setServiceName(kafkaServicesList);
+              return serviceIntegration;
+            });
+            serviceIntegrations.forEach((i) => {
```
Comment (mohammed-zahaf) [inline packages/manager/modules/pci/src/projects/project/storages/databases/database/service-integration/service-integration.routing.js:52]:
```suggestion
            serviceIntegrations.forEach((serviceIntegration) => {
```

File: packages/manager/modules/pci/src/projects/project/storages/databases/database/service-integration/service-integration.routing.js [hunk 5]
```diff
@@ -0,0 +1,127 @@
...
+        ) => () =>
+          serviceIntegrationList.forEach((s) =>
+            DatabaseService.stopPollingIntegrationStatus(database.id, s.id),
+          ),
+        addableServicesList: /* @ngInject */ (
+          kafkaServicesList,
+          serviceIntegrationList,
+        ) =>
+          kafkaServicesList.filter(
+            (k) => !serviceIntegrationList.find((i) => i.serviceId === k.id),
```
Comment (mohammed-zahaf) [inline packages/manager/modules/pci/src/projects/project/storages/databases/database/service-integration/service-integration.routing.js:84]:
```suggestion
            ({id}) => !serviceIntegrationList.find(({serviceId}) => id === serviceId),
```

File: packages/manager/modules/pci/src/projects/project/storages/databases/database/service-integration/translations/Messages_fr_FR.json [hunk 6]
```diff
@@ -0,0 +1,9 @@
+{
+  "pci_databases_service_integration_tab_title": "Intégration de services",
+  "pci_databases_service_integration_tab_descritpion": "L’intégration de services vous permet de connecter différentes solutions Kafka.",
+  "pci_databases_service_integration_tab_kafka_service": "Service Kafka",
+  "pci_databases_service_integration_tab_add_kafka": "Ajouter un Kafka",
```
Comment (mohammed-zahaf) [inline packages/manager/modules/pci/src/projects/project/storages/databases/database/service-integration/translations/Messages_fr_FR.json:5]:
```suggestion
  "pci_databases_service_integration_tab_add_kafka": "Ajouter un service Kafka",
```

File: packages/manager/modules/pci/src/projects/project/storages/databases/database/service-integration/translations/Messages_fr_FR.json [hunk 7]
```diff
@@ -0,0 +1,9 @@
+{
+  "pci_databases_service_integration_tab_title": "Intégration de services",
```
Comment (mohammed-zahaf) [inline packages/manager/modules/pci/src/projects/project/storages/databases/database/service-integration/translations/Messages_fr_FR.json:2]: I think it's better:
```suggestion
  "pci_databases_service_integration_tab_title": "Intégration de service Kafka",
```

File: packages/manager/modules/pci/src/projects/project/storages/databases/database/service-integration/translations/Messages_fr_FR.json [hunk 8]
```diff
@@ -0,0 +1,9 @@
...
+  "pci_databases_service_integration_tab_descritpion": "L’intégration de services vous permet de connecter différentes solutions Kafka.",
+  "pci_databases_service_integration_tab_kafka_service": "Service Kafka",
+  "pci_databases_service_integration_tab_add_kafka": "Ajouter un Kafka",
+  "pci_databases_service_integration_tab_no_kafka_service": "Vous ne pouvez pas ajouter un Kafka car vous n'avez plus d'autre service Kafka disponible à connecter.",
+  "pci_databases_service_integration_tab_unknown_service": "Service non trouvé",
+  "pci_databases_service_integration_tab_service_ready": "L'intégration de votre service Kafka est prête"
```
Comment (mohammed-zahaf) [inline packages/manager/modules/pci/src/projects/project/storages/databases/database/service-integration/translations/Messages_fr_FR.json:8]:
```suggestion
  "pci_databases_service_integration_tab_service_ready": "L'intégration de votre service Kafka est prêt"
```

File: packages/manager/modules/pci/src/projects/project/storages/databases/database/service-integration/add/add.html (hunk 1 above)
Comment (jperchoc) [reply packages/manager/modules/pci/src/projects/project/storages/databases/database/service-integration/add/add.html:2]: As suggested in a previous PR, I removed the form elements when we do not have any submit (this is handled by the oui-modal action attributes. However, we still need the ng-form directive to activate input validation

File: packages/manager/modules/pci/src/components/project/storages/databases/serviceIntegration.class.js [hunk 9]
```diff
@@ -0,0 +1,22 @@
+import find from 'lodash/find';
+import Base from './base.class';
```
[the remark this answers is not in the document]
Comment (marie-j) [reply packages/manager/modules/pci/src/components/project/storages/databases/serviceIntegration.class.js:2]: Maybe could be planned as a refactoring ? If we get the chance to improve why not take it :)

File: packages/manager/modules/pci/src/components/project/storages/databases/serviceIntegration.class.js [hunk 10]
```diff
@@ -0,0 +1,22 @@
...
+export default class ServiceIntegration extends Base {
+  constructor({ id, serviceId, status, type }) {
+    super();
+    this.updateData({
+      id,
+      serviceId,
+      status,
+      type,
+    });
+  }
```
[the remark this answers is not in the document]
Comment (marie-j) [reply packages/manager/modules/pci/src/components/project/storages/databases/serviceIntegration.class.js:13]: But if we have a look at the rest of the codebase, this is not the way it is done so we have consistency but scoped to a folder

File: packages/manager/modules/pci/src/components/project/storages/databases/serviceIntegration.class.js [hunk 11]
```diff
@@ -0,0 +1,22 @@
...
+
+  setServiceName(serviceList) {
+    this.serviceName = find(serviceList, { id: this.serviceId })?.description;
+  }
```
[the remark this answers is not in the document]
Comment (jperchoc) [reply packages/manager/modules/pci/src/components/project/storages/databases/serviceIntegration.class.js:17]: This has been reworked here: https://github.com/ovh/manager/pull/5865

File: packages/manager/modules/pci/src/projects/project/storages/databases/database/service-integration/add/translations/Messages_fr_FR.json (hunk 2 above)
Comment (jperchoc) [reply packages/manager/modules/pci/src/projects/project/storages/databases/database/service-integration/add/translations/Messages_fr_FR.json:4]: I cannot change the text as it has been already validated by content team.

File: packages/manager/modules/pci/src/projects/project/storages/databases/database/service-integration/delete/translations/Messages_fr_FR.json (hunk 3 above)
Comment (jperchoc) [reply packages/manager/modules/pci/src/projects/project/storages/databases/database/service-integration/delete/translations/Messages_fr_FR.json]: Corrected here: https://github.com/ovh/manager/commit/5cea0769b5533ded62dec47a2dba10ef635a0290

File: packages/manager/modules/pci/src/projects/project/storages/databases/database/service-integration/translations/Messages_fr_FR.json (hunk 6 above)
Comment (jperchoc) [reply packages/manager/modules/pci/src/projects/project/storages/databases/database/service-integration/translations/Messages_fr_FR.json:5]: Same: I cannot change the text as it has been already validated by content team.

File: packages/manager/modules/pci/src/projects/project/storages/databases/database/service-integration/translations/Messages_fr_FR.json (hunk 7 above)
Comment (jperchoc) [reply packages/manager/modules/pci/src/projects/project/storages/databases/database/service-integration/translations/Messages_fr_FR.json:2]: same: I cannot change the text as it has been already validated by content team.

File: packages/manager/modules/pci/src/projects/project/storages/databases/database/service-integration/translations/Messages_fr_FR.json (hunk 8 above)
Comment (jperchoc) [reply packages/manager/modules/pci/src/projects/project/storages/databases/database/service-integration/translations/Messages_fr_FR.json:8]: same: I cannot change the text as it has been already validated by content team. Here it is the integration that is ready, so the text is ok

File: packages/manager/modules/pci/src/components/project/storages/databases/serviceIntegration.class.js (hunk 10 above)
[the remark this answers is not in the document]
Comment (jperchoc) [reply packages/manager/modules/pci/src/components/project/storages/databases/serviceIntegration.class.js:13]: Then this is something that should be addressed during the refactoring

File: packages/manager/modules/pci/src/projects/project/storages/databases/database/service-integration/service-integration.html [hunk 12]
```diff
@@ -0,0 +1,92 @@
...
+                <span
+                    data-translate="pci_databases_service_integration_tab_add_kafka"
+                ></span>
+            </oui-button>
+            <button
+                data-ng-if="$ctrl.serviceIntegrationList.length > 0"
+                class="float-right oui-button oui-button_secondary oui-button_icon-only oui-button_s"
+                type="button"
+                data-title="{{:: 'pci_databases_common_refresh' | translate }}"
+                data-ng-click="$ctrl.refreshing=true; $ctrl.refreshServiceIntegration()"
```
Comment (mohammed-zahaf) [inline packages/manager/modules/pci/src/projects/project/storages/databases/database/service-integration/service-integration.html:82]: More than one action have to be implemented into a dedicated function.

```suggestion
                data-ng-click="$ctrl.onRefreshServiceIntegrationClick()"
```

**Review by mohammed-zahaf — APPROVED** [review]

File: packages/manager/modules/pci/src/projects/project/storages/databases/database/service-integration/service-integration.html (hunk 12 above)
Comment (jperchoc) [reply packages/manager/modules/pci/src/projects/project/storages/databases/database/service-integration/service-integration.html]: Changed here: https://github.com/ovh/manager/pull/5765/commits/d401f3622207c4c8282bc3619d16efe2d2210000

**Review by antleblanc — APPROVED** [review]

**Pull request merged**